src/containers/RapidAppointment/RapidAppointment.tsx

```diff
@@ -1,9 +1,641 @@
+      setAddress(locationData.address?.line1 || "");
+      setCity(locationData.address?.city || "");
+      setProvince(locationData.address?.province || "");
+      setPostalCode(locationData.address?.postcode || "");
+      setPhoneNumber(locationData.phone || "");
+      setWebsite(locationData.url || "");
+      setShouldShowExpandedForm(false);
+    }
+  }, [locationData, previousData]);
+
+  const validateForm = () => {
+    let isValid = true;
+    if (!name) {
+      setShouldShowInvalidName(true);
+      isValid = false;
+    } else {
+      setShouldShowInvalidName(false);
+    }
+
+    if (!address) {
+      setShouldShowInvalidAddress(true);
+      isValid = false;
+    } else {
+      setShouldShowInvalidAddress(false);
+    }
+
+    if (!city) {
+      setShouldShowInvalidCity(true);
+      isValid = false;
+    } else {
+      setShouldShowInvalidCity(false);
+    }
+
+    if (!province) {
+      setShouldShowInvalidProvince(true);
+      isValid = false;
+    } else {
+      setShouldShowInvalidProvince(false);
+    }
+
+    if (!postalCodeIsValid(postalCode)) {
+      setShouldShowInvalidPostal(true);
+      isValid = false;
+    } else {
+      setShouldShowInvalidPostal(false);
+    }
+
+    if (!website) {
+      setShouldShowInvalidURL(true);
+      isValid = false;
+    } else {
+      setShouldShowInvalidURL(false);
+    }
+
+    if (
+      !isWalkInChecked &&
+      !isEmailChecked &&
+      !isCallAheadChecked &&
+      !isVisitWebsiteChecked
+    ) {
+      setShouldShowInvalidBooking(true);
+      isValid = false;
+    } else {
+      setShouldShowInvalidBooking(false);
+    }
+
+    if (!isCancellationsChecked && !isExpiringDosesChecked) {
+      setShouldShowInvalidReasons(true);
+      isValid = false;
+    } else {
+      setShouldShowInvalidReasons(false);
+    }
+    return isValid;
+  };
+
+  const handleSubmit = () => {
+    if (!validateForm()) {
+      return;
+    }
+
+    const tagsCommaSeparatedString: string[] = [];
+
+    if (isCallAheadChecked) {
+      tagsCommaSeparatedString.push("Call Ahead");
+    }
+    if (isWalkInChecked) {
+      tagsCommaSeparatedString.push("Walk In");
+    }
+
+    if (isVisitWebsiteChecked) {
+      tagsCommaSeparatedString.push("Visit Website");
+    }
+    if (isEmailChecked) {
+      tagsCommaSeparatedString.push("Email");
+    }
+
+    if (isCancellationsChecked) {
+      tagsCommaSeparatedString.push("Cancellation");
+    }
+
+    if (isExpiringDosesChecked) {
+      tagsCommaSeparatedString.push("Expiring Doses");
+    }
+
+    const utcDate = zonedTimeToUtc(
+      startOfDay(new Date()),
+      Intl.DateTimeFormat().resolvedOptions().timeZone,
+    );
+
+    const requestPayload: VaccineAvailabilityExpandedCreateRequest = {
+      active: 1, // boolean indicating if location is active
+      date: format(utcDate, "yyyy-MM-dd'T'HH:mm:ssxxx"),
+      inputType: 1, // represents how availability data was recorded - not used at time of writing
```

Comment:
Let's change this to 2 so we can query for inputType 2 for the removal function

Reply:
Updated